rust/arrow/src/array/array_primitive.rs

```diff
@@ -112,6 +112,22 @@ impl<T: ArrowPrimitiveType> PrimitiveArray<T> {
         );
         PrimitiveArray::from(data)
     }
+
+    /// Creates a PrimitiveArray based on a constant value with `count` elements
+    pub fn from_value(value: T::Native, count: usize) -> Self {
+        // # Safety: length is known
+        let val_buf = unsafe { Buffer::from_trusted_len_iter((0..count).map(|_| value)) };
```

Comment:
@Dandandan have you tried `iter::repeat(value).take(count)` ? wouldn't that be a better fit in this case (because the values in the (0..count) range are not used) ?

Reply:
`take` implementation also needs a counter value to limit the nr of values. https://doc.rust-lang.org/src/core/iter/adapters/take.rs.html#35
So ot seemed to me slighty easier implementation (but repeat + take is also looks fine to me, you could argue that that is a better fit as well). As you can see it was using `repeat(*value).take(size))` in an earlier version in datafusion.

I'm not sure if there is any difference in compiled code, can run it through godbolt to see what is changed. I think `iter::repeat(value).take(count)` would be slightly harder to optimize to a efficient loop, but could very well be that the generated code is very similar.